### PR TITLE
Fix bad interaction between hold to attack and quick cast

### LINKED ISSUE #4765: Bad interaction between hold to attack and quick-cast
**Important information**
Linux / 127c6bc9e1b395982ce953cbfca3e8dca05ca949

**Describe**
Quick casting while holding LMB to attack causes the readied spell to be cast continually.

**To Reproduce**
1. Enable quick cast.
2. Load up a character with at least 2 spells. Let's call them A and B
3. Assign a hotkey to one spell, and ready another.
4. Hold shift-click to attack
5. Quick cast the former spell

**Expected behavior**
The character should continue auto-attacking after the quick cast.

### PR DESCRIPTION
**Pull request by burningserenity**
Fixes #4765

I thought I was going to look in the controls to handle repeat mouse actions, but the answer remained in the player controls.
I had to set the quick cast last mouse button values specifically to attack values, because the last mouse button action is always
set to a spell action on casting a spell in player.cpp.

Tested with keyboard and mouse only at this point.

### CODE REVIEW

**Comment by burningserenity** [comment]
Oops. Committed some extra code.

**Comment by julealgon** [comment]
@burningserenity you might want to keep your PR in draft mode while you are still working on it. Reduces notification noise a bit.

**Comment by burningserenity** [comment]
Thanks, I'll do that next time :-)

PR is ready for review, in any case.

**Comment by AJenbo** [comment]
It also looks like you are buliding this on top of your previous branch rather then startin a new one?

**Comment by AJenbo** [comment]
Well it's mostly that you now have 17 commits to this branch so reading the history is a bit messy. But as long as you aren't making multiple changes in on PR we can just squash merge it in to a single commit.

File: Source/controls/plrctrls.cpp [hunk 1]
```diff
@@ -1991,6 +1991,7 @@ void PerformSecondaryAction()
 
 void QuickCast(size_t slot)
 {
+	MouseActionType PrevMouseButtonAction = LastMouseButtonAction;
```
Comment (AJenbo) [inline Source/controls/plrctrls.cpp:1994]: We use uppercase for globals and lowercase for locals :)

```suggestion
	MouseActionType prevMouseButtonAction = LastMouseButtonAction;
```

**Comment by burningserenity** [comment]
Should I just leave these as drafts until they're approved? This is still an awful lot of commits to notify everyone about.

**Comment by AJenbo** [comment]
No, you gave gotten to the review state so it shouldn't be a draft :) Draft is basically for when you just want feedback from the CI but not necessarily humans.

**Comment by AJenbo** [comment]
Looks like you made the type lowercase instead of the variable.

**Review by AJenbo — APPROVED** [review]

**Pull request merged**